README.md

```diff
@@ -1,2 +1,69 @@
-# jenkins-library
-PCIC Jenkins shared libary
+# PCIC Jenkins Pipeline Shared Library
+This repo contains the shared library for Jenkins.
+
+## Setup in Jenkins
+To get access to this library you have to let Jenkins know about it.
+
+1. Navigate to **Manage Jenkins > Configure System > Global Pipeline Libraries**
+2. Fill in the following sections:
+    - **Name:** `pcic-pipeline-library`
+    - **Default Version:** `master`
+    - **Retrieval Method:**: `Modern SCM`
+    - **Source Code Management:** `Git`
+    - **Project Repository:** `https://github.com/pacificclimate/jenkins-library.git`
+3. **Save**
+
+To use the library steps in a Jenkinsfile add the following line to the top of your file:
+```
+@Library('pcic-pipeline-library')_
+```
+**NOTE:** The `_` is not a typo and must be included.
+
+## Supported Steps
+The following table shows all of the currently supported steps.
+
+| Step | Description |
+|------|-------------|
+| [`buildDockerImage`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/buildDockerImage.groovy) | Build a Docker image onto a Docker server. |
+| [`codeCollection`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/codeCollection.groovy) | Uses the standard `checkout scm` command (only available for multi-branch pipelines) along with a git fetch. |
+| [`publishDockerImage`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/publishDockerImage.groovy) | Publishes a docker image to a registry.  By default the name of the image should contain the registry to publish to in the form `your-registry/image-name:tag`.  There is an option to specify a registry with a URL. |
+| [`publishPythonPackage`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/publishPythonPackage.groovy) | Creates a python package out of the project and publishes it to pypi. |
+| [`pypiPublishable`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/pypiPublishable.groovy) | Checks branch conditions to see if the python package should be published. |  
+| [`removeDockerImage`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/removeDockerImage.groovy) | Removes the built Docker image from the server. |
+| [`runNodeTestSuite`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/runNodeTestSuite.groovy) | Runs a node test suite. |
+| [`runPythonTestSuite`](https://github.com/pacificclimate/jenkins-library/blob/dev/vars/runPythonTestSuite.groovy) | Runs a python test suite. |
+
+
+
+## Structure
+This section outlines how the project is strucuted. If you are looking to add functionality please read this section before coding.  
+
+### Steps
+All of the steps are located in the [`vars/`](https://github.com/pacificclimate/jenkins-library/tree/dev/vars) directory.  Each `.groovy` file corresponds to a step where the step name is the name of the file.  This is the name you would use in a `Jenkinsfile` to use the step.  In the file there **must** be a `call` function:
```

Comment:
I agree with your comment but it is the convention.  `src`, `test` and `vars` are meant to be top level directories.  As for the `org/pcic` path, I've seen some differences in my exploration of other libraries.  Sometimes people use `src/com/yourcompany`, others use `src/org/yourcompany` and I've even seen deeper paths like `src/com/groovy/yourcompany`.  We _could_ break with convention and maybe group the classes into more descriptive package names like `src/classes` and `src/util`.

Reply:
Until I know why the convention exists, I'd hesitate to break it. Although the convention looks a little odd and heavy-handed to me, it might be best to keep it if it is widely followed.